metacat-main/src/main/java/com/netflix/metacat/main/configs/ElasticSearchConfig.java

```diff
@@ -27,34 +27,41 @@
 import com.netflix.metacat.main.services.DatabaseService;
 import com.netflix.metacat.main.services.PartitionService;
 import com.netflix.metacat.main.services.TableService;
+import com.netflix.metacat.main.services.search.ElasticSearchEventHandlers;
 import com.netflix.metacat.main.services.search.ElasticSearchMetric;
 import com.netflix.metacat.main.services.search.ElasticSearchRefresh;
 import com.netflix.metacat.main.services.search.ElasticSearchUtil;
 import com.netflix.metacat.main.services.search.ElasticSearchUtilImpl;
-import com.netflix.metacat.main.services.search.ElasticSearchEventHandlers;
 import com.netflix.spectator.api.Registry;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang.StringUtils;
 import org.elasticsearch.client.Client;
 import org.elasticsearch.client.transport.TransportClient;
-import org.elasticsearch.common.settings.ImmutableSettings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
 /**
  * Configuration for ElasticSearch which triggers when metacat.elasticsearch.enabled is true.
  *
  * @author tgianos
+ * @author zhenl
  * @since 1.1.0
  */
-//TODO: Look into spring data elasticsearch to replace this
+@Slf4j
 @Configuration
 @ConditionalOnProperty(value = "metacat.elasticsearch.enabled", havingValue = "true")
 public class ElasticSearchConfig {
 
+    private static final String TCP_CONNECTION_TIMEOUT = "60s";
```

Comment:
With this being used only once maybe you should just have it as a local variable? Additionally this looks like something that would want to be a property with a default value.

Reply:
Ok. Don't think we need to make it configurable. Roll back the code to previous.